x/warp/config.go

```diff
@@ -118,7 +120,8 @@ func (c *Config) verifyWarpMessage(predicateContext *precompileconfig.PredicateC
 	// Verify the warp payload can be decoded to the expected type
 	_, err := warpPayload.ParseAddressedPayload(warpMsg.UnsignedMessage.Payload)
 	if err != nil {
-		return fmt.Errorf("%w: %s", errInvalidAddressedPayload, err)
+		log.Debug("failed to parse addressed payload", "msgID", warpMsg.ID(), "payload", warpMsg.UnsignedMessage.Payload, "err", err)
```

Comment:
Why don't we need to check for the block hash payload as a possibility here?

Reply:
Dropping this. The contract execution will parse it and handle failure to parse as a failure.